### PR TITLE
Feature/dish modal

### PR DESCRIPTION
**Pull request by OrenMoveo**
# Dish Modal Feature

## This PR Includes
- Responsive UI design for both mobile and desktop platforms.
- Implementation of a portal that renders the modal as a sibling container of the root.
- Context implementation to prevent the need for prop propagation.
- Transformation of each dish into a modal upon clicking.
- Application of a constant max width to manage the 'stretchy' behavior on larger screens.
- On mobile devices, clicking a modal scrolls it to the top of the window.

## Notes
- A design decision i made ensures that the hero and navbar consistently display at 100% of the screen width.
- Prettier configuration has been updated based on Rabea's advice, extending the maximum line length from 80 to 180 words per line. As a result, some files underwent changes, primarily in indentation, including the 'menuPopover' component.

## Example

https://github.com/OrenMoveo/Epicure/assets/156906187/ce5cf9cf-9c9f-41db-83c6-135f2606cb27

### CODE REVIEW

File: src/components/ChooseQuantity/ChooseQuantity.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  const handleDecreaseQuantity = (prevQuantity: number) => {
+    if (prevQuantity === 1) {
+      return;
+    }
+    setQuantity(prevQuantity - 1);
+  };
+
+  const handleIncreaseQuantity = (prevQuantity: number) => {
+    setQuantity(prevQuantity + 1);
```
Comment (nitzanpap) [inline src/components/ChooseQuantity/ChooseQuantity.tsx:17]: All these `1`s are magic numbers. I do not understand what they mean, not will you in a week from now.
Please save them into constant variables with a meaningful name/s.

File: src/components/DishCard/DishCard.tsx [hunk 2]
```diff
@@ -18,61 +21,49 @@ interface DishCardProps {
...
 const DishCard: FC<DishCardProps> = (props) => {
+  const { updateDish, setIsModalActive } = useModalContext();
+  const isMobile = useIsMobile();
+  const isTablet = useIsTablet();
+  const isMobileOrTable = isMobile || isTablet;
+  const handleModalClick = () => {
+    setIsModalActive(true);
+    updateDish(props.dish);
+    if (isMobileOrTable) window.scrollTo(0, 0);
+  };
```
Comment (nitzanpap) [inline src/components/DishCard/DishCard.tsx:36]: Adding a comment here, or exporting this functionality into another function with a meaningful name would result in a cleaner code.

File: src/components/DishModal/DishModal.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                <ChooseQuantity />
+              </div>
+              <button className={styles.addToBagBtn}>ADD TO BAG</button>
+            </div>
+          </div>
+        </section>
+        {isMobileOrTablet && <Footer />}
+      </div>
+    </div>,
+    document.getElementById("modal")!
```
Comment (nitzanpap) [inline src/components/DishModal/DishModal.tsx:153]: This exclamation mark could result in a hard-to-find bug. What happens if for some reason this `#modal` element is not found? Please test that scenario, and handle (if needed) any resulting bug.

File: src/context/ModalContext/ModalContext.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const defaultDish = {
+  name: "Pad Ki Mao",
+  pictureUrl: "/src/assets/images/dishes/padKiMao.png",
+  description: "Green Papaya, Mango, Chukka Chili, Mint, Kaffir lime, Cashew, Akaya Cham sauce",
+  price: 88,
+  signatureDish: true,
+  foodIcon: "/src/assets/images/spicyIcon.svg",
+  mealType: "dinner",
+  keyId: "9",
+};
```
Comment (nitzanpap) [inline src/context/ModalContext/ModalContext.tsx:12]: Why do you keep this mock data here? Please export this to somewhere else.

**Review by nitzanpap — CHANGES_REQUESTED** [review]
You can merge this PR after you fix all the issues. You may also click "resolve" on each of them once you have finished them, instead of waiting for me.

File: src/components/ChooseQuantity/ChooseQuantity.tsx (hunk 1 above)
Comment (OrenMoveo) [reply src/components/ChooseQuantity/ChooseQuantity.tsx:17]: @nitzanpap  done

File: src/components/DishCard/DishCard.tsx (hunk 2 above)
Comment (OrenMoveo) [reply src/components/DishCard/DishCard.tsx:36]: @nitzanpap  done

File: src/components/DishModal/DishModal.tsx (hunk 3 above)
Comment (OrenMoveo) [reply src/components/DishModal/DishModal.tsx:153]: @nitzanpap  done

File: src/context/ModalContext/ModalContext.tsx (hunk 4 above)
Comment (OrenMoveo) [reply src/context/ModalContext/ModalContext.tsx:12]: @nitzanpap  done

**Pull request closed without merging**